lnwallet/channel.go

```diff
@@ -2617,6 +2569,64 @@ func (lc *LightningChannel) evaluateHTLCView(view *htlcView, ourBalance,
 	return newView, nil
 }
 
+// fetchParent is a function which is used to look up an update's parent in
+// the appropriate update log.
+type fetchParent func(entry *PaymentDescriptor,
+	remoteLog bool) (*PaymentDescriptor, error)
+
+// getFetchParent is a helper that returns a function for looking up parent
+// entries in the appropriate log.
+func (lc *LightningChannel) getFetchParent(remoteChain bool) fetchParent {
+	return func(entry *PaymentDescriptor,
+		remoteLog bool) (*PaymentDescriptor, error) {
+
+		var (
+			updateLog *updateLog
+			logName   string
+		)
+
+		if remoteLog {
+			updateLog = lc.remoteUpdateLog
+			logName = "remote"
+		} else {
+			updateLog = lc.localUpdateLog
+			logName = "local"
+		}
+
+		addEntry := updateLog.lookupHtlc(entry.ParentIndex)
+
+		switch {
+		// We check if the parent entry is not found at this point.
+		// This could happen for old versions of lnd, and we return an
+		// error to gracefully shut down the state machine if such an
+		// entry is still in the logs.
+		case addEntry == nil:
```

Comment:
these three cases are still uncovered, do we want to test them in this pr?

Reply:
we do indeed! Added them in a separate fetchParent test bc it's easier to split up that way.

Reply:
awesome, nice addition! looking at coveralls the middle case `case remoteChain && addEntry.addCommitHeightRemote == 0:` is still uncovered, maybe just a configuration issue in the unit test?

Reply:
Local and remote log values mixed up in the `remote log, remote add height 0` case, fixed :)